### PR TITLE
Gitpodified the repo

### PR DESCRIPTION
**Pull request by JanKoehnlein**
Hi buildpacks team! 👋

here is a PR that fully-automates the dev setup of buildpacks/lifecycle with Gitpod, providing anyone with a pre-built, browser-based development environment in one click:

[Open in Gitpod](https://gitpod.io/#https://github.com/buildpacks/lifecycle/pull/604)

Screen Shot 2021-05-04 at 18 01 53

### What it does

* Comes with all buildpack lifecycle dependencies pre-installed (Go, Docker...)
* Runs `make  build` ahead-of-time so you don't need to wait
* Gives anyone immediate access to the pre-compiled buildpacks CLI, and allows to easily edit & rebuild the code

I hope this can make life easier for contributors, and for maintainers who need to review many PRs (hint: you can already open this PR in Gitpod to easily review & test it without messing up your local checkout).

I'm looking forward to your feedback on this PR! 🚀

### CODE REVIEW

File: .gitpod.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+tasks:
+  - init: make build
```
Comment (jabrown85) [inline .gitpod.yml:2]:
```suggestion
  - init: make build-linux
```
We only really need to build linux binaries for something like this.

File: tools/go.mod [hunk 2]
```diff
@@ -9,7 +9,7 @@ require (
 	github.com/docker/docker v1.4.2-0.20190924003213-a8608b5b67c7
 	github.com/golang/mock v1.5.0
 	github.com/golangci/golangci-lint v1.30.0
-	github.com/google/go-containerregistry v0.4.1
+	github.com/google/go-containerregistry v0.5.0
```
Comment (jabrown85) [inline tools/go.mod:12]: This change seems unrelated

File: .gitpod.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+github:
+  prebuilds:
+    master: true
```
Comment (jabrown85) [inline .gitpod.yml:5]: Does this need to be `main`? Or is `main` really mean default branch?

**Review by jabrown85 — COMMENTED** [review]
This looks neat!

File: .gitpod.yml (hunk 3 above)
Comment (JanKoehnlein) [reply .gitpod.yml:5]: `master` means `default` branch here (for historic reasons)

File: .gitpod.yml (hunk 1 above)
Comment (JanKoehnlein) [reply .gitpod.yml]: OK

File: tools/go.mod (hunk 2 above)
Comment (JanKoehnlein) [reply tools/go.mod]: Good catch. Reverted

**Comment by JanKoehnlein** [comment]
Thanks a lot for your feedback. I've addressed your comments and added the Go extension for the IDE to be loaded automatically.

File: go.sum [hunk 4]
```diff
@@ -155,8 +155,8 @@ github.com/google/go-cmp v0.5.0/go.mod h1:v8dTdLbMG2kIc/vJvl+f65V22dbkXbowE6jgT/
 github.com/google/go-cmp v0.5.2/go.mod h1:v8dTdLbMG2kIc/vJvl+f65V22dbkXbowE6jgT/gNBxE=
 github.com/google/go-cmp v0.5.5 h1:Khx7svrCpmxxtHBq5j2mp/xVjsi8hQMfNLvJFAlrGgU=
 github.com/google/go-cmp v0.5.5/go.mod h1:v8dTdLbMG2kIc/vJvl+f65V22dbkXbowE6jgT/gNBxE=
-github.com/google/go-containerregistry v0.4.1 h1:Lrcj2AOoZ7WKawsoKAh2O0dH0tBqMW2lTEmozmK4Z3k=
 github.com/google/go-containerregistry v0.4.1/go.mod h1:Ct15B4yir3PLOP5jsy0GNeYVaIZs/MK/Jz5any1wFW0=
+github.com/google/go-containerregistry v0.5.0 h1:eb9sinv4PKm0AUwQGov0mvIdA4pyBGjRofxN4tWnMwM=
```
Comment (jabrown85) [inline go.sum:159]: I think this are probably not your fault, but something on `main` is out of date. Can you remove these from the PR though?

File: tools/go.sum [hunk 5]
```diff
@@ -244,8 +244,9 @@ github.com/google/go-cmp v0.5.0/go.mod h1:v8dTdLbMG2kIc/vJvl+f65V22dbkXbowE6jgT/
 github.com/google/go-cmp v0.5.2/go.mod h1:v8dTdLbMG2kIc/vJvl+f65V22dbkXbowE6jgT/gNBxE=
 github.com/google/go-cmp v0.5.5 h1:Khx7svrCpmxxtHBq5j2mp/xVjsi8hQMfNLvJFAlrGgU=
 github.com/google/go-cmp v0.5.5/go.mod h1:v8dTdLbMG2kIc/vJvl+f65V22dbkXbowE6jgT/gNBxE=
-github.com/google/go-containerregistry v0.4.1 h1:Lrcj2AOoZ7WKawsoKAh2O0dH0tBqMW2lTEmozmK4Z3k=
 github.com/google/go-containerregistry v0.4.1/go.mod h1:Ct15B4yir3PLOP5jsy0GNeYVaIZs/MK/Jz5any1wFW0=
+github.com/google/go-containerregistry v0.5.0 h1:eb9sinv4PKm0AUwQGov0mvIdA4pyBGjRofxN4tWnMwM=
```
Comment (jabrown85) [inline tools/go.sum:248]: Same here for these. Please remove

**Comment by jabrown85** [comment]
> Go extension for the IDE 

How did you set that up? I don't see it in the `.gitpod.yml` and I don't see it when opening up with the link in the PR description.

image

File: go.sum (hunk 4 above)
Comment (JanKoehnlein) [reply go.sum]: OK, done

File: tools/go.sum (hunk 5 above)
Comment (JanKoehnlein) [reply tools/go.sum]: Done.

**Comment by JanKoehnlein** [comment]
I've removed all changes form the `go.sum` files. Note that I had these in place to make the build pass, so somebody might have to investigate the failures.
Into the bargain, I _really_ added the Go and Dockerfile extension to the config this time.

**Comment by jabrown85** [comment]
@JanKoehnlein thanks for the update - The button in the PR description opens up the `main` branch and not this PR. I tried making a link like https://gitpod.io/#https://github.com/buildpacks/lifecycle/-/tree/JanKoehnlein:jk/gitpodify and https://gitpod.io/#https://github.com/buildpacks/lifecycle/-/merge_requests/604 but neither seemed to work like I expected. I was hoping to see the first build attempt and the extensions get properly installed.

Do you know how to make the link properly?

**Comment by JanKoehnlein** [comment]
In Gitpod it's always the URL of the GitHub/Gitlab/Bitbucket context prefixed with `gitpod.io/#` 
So here it's (`https://gitpod.io/#https://github.com/buildpacks/lifecycle/pull/604`)[https://gitpod.io/#https://github.com/buildpacks/lifecycle/pull/604]

I've updated the button in the description above. Note that to have pre-builds automatically created, you must enable the (Gitpod GitHub app)[https://www.gitpod.io/docs/prebuilds] for your repo.

**Review by jabrown85 — APPROVED** [review]

**Review by yaelharel — APPROVED** [review]

**Review by natalieparellano — APPROVED** [review]
